pkg/controller/blockdeviceclaim/blockdeviceclaim_controller.go

```diff
@@ -205,14 +213,18 @@ func (r *ReconcileBlockDeviceClaim) FinalizerHandling(
 	if len(instance.ObjectMeta.Finalizers) == 1 &&
 		util.Contains(instance.ObjectMeta.Finalizers, controllerutil.BlockDeviceClaimFinalizer) {
 		// Finalizer is set, lets handle external dependency
-		if err := r.releaseClaimedBlockDevice(instance, reqLogger); err != nil {
+		bd,err := r.releaseClaimedBlockDevice(instance, reqLogger)
+		if err != nil {
 			reqLogger.Error(err, "Could not delete external dependency", "BlockDeviceClaim-CR:", instance.ObjectMeta.Name)
 			return err
 		}
+		r.recorder.Eventf(bd,corev1.EventTypeNormal,"BlockDeviceCleanUpInProgress","CleanUp ")
```

Comment:
Need to change `Cleanup` to `Released`. The cleanup related events can occur only from the blockdevice controller. The blockdeviceclaim controller marks the BD as Released, so that the BD controller can take it up for cleaning.